app/services/hashtag_autocomplete_service.rb

```diff
@@ -114,10 +123,16 @@ def initialize(params = {})
       @ref = params[:ref]
       @slug = params[:slug]
       @id = params[:id]
+
+      if @type == "category"
```

Comment:
I would really like to keep any type-specific conditionals out of here, all of the hashtag stuff is set up to avoid this :)

In this case, IMO just allow any of them, but only set `style_icon` and `style_emoji` if `style_type` is not null.

At some point, chat may want to use these as well for the channel icons like we discussed (since ATM if you've set an icon for the category it doesn't show for the channel)